tests/redirects_tests/test_middleware.py

```diff
@@ -0,0 +1,131 @@
+from django.contrib.redirects.middleware import RedirectFallbackMiddleware
+from django.contrib.redirects.models import Redirect
+from django.http import (
+    HttpResponse, HttpResponseForbidden, HttpResponseNotFound,
+    HttpResponseRedirect,
+)
+from django.test import (
+    RequestFactory, TestCase, modify_settings, override_settings,
+)
+
+
+@modify_settings(ALLOWED_HOSTS={'append': 'test.loc'})
+@override_settings(APPEND_SLASH=False)
+class RedirectFallbackMiddlewareTests(TestCase):
+    def setUp(self):
+        self.middleware = RedirectFallbackMiddleware(
+            lambda r: HttpResponseNotFound()
+        )
+        self.rf = RequestFactory()
+
+    def test_blank_redirect(self):
+        r = Redirect.objects.create(domain='', old_path='/old', new_path='/new')
+        request = self.rf.get(r.old_path)
+
+        response = self.middleware(request)
+
+        self.assertRedirects(response, r.new_path,
+                             status_code=301, fetch_redirect_response=False)
```

Comment:
This should use longer lines (up to 119 characters) or hanging indent as described in [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).